public/lib/uploadHttpConcurrentProgress.js

```diff
@@ -116,8 +116,11 @@
             return;
         }
         this._collectMovingAverages = false;
-        //pushing results to an array
-        this._results.push(result);
+
+        //if request complete and no progress events then lower movingAverage
+        if(this._progressCount < this.movingAverage){
+          this.movingAverage =1;
```

Comment:
logic removed ... updated to report result to client and store result .. addresses low bandwidth measurements